src/collections/map.rs

```diff
@@ -53,6 +58,16 @@ where
     V: State<S>,
     S: Read,
 {
+    pub fn contains_key(&self, key: K) -> Result<bool> {
+        let child_contains = self.children.contains_key(&key);
```

Comment:
This could be optimized to prevent an unnecessary read from the store by returning early if `child_contains` is true, rather than just checking it at the end.

Reply:
Good catch. I saw the opportunity for a clean return statement and didn't think about the store access overhead. I'll keep that in mind.